internal/resource/default_user_secret.go

```diff
@@ -146,6 +147,18 @@ func (builder *DefaultUserSecretBuilder) updatePorts(secret *corev1.Secret) {
 	}
 }
 
+// generateUsername returns a base64 string that has "default_user_" as prefix
+// returned string has length 'l' when base64 decoded
+func generateUsername(l int) (string, error) {
+	encodedStr, err := randomEncodedString(l)
+	if err != nil {
+		return "", err
+	}
+
+	encodedStrSlice := []byte(encodedStr)
+	return string(append([]byte(usernamePrefix), encodedStrSlice[0:len(encodedStrSlice)-len(usernamePrefix)]...)), nil
+}
+
```

Comment:
I think it's a bit clearer to read when directly generating the random string with the correct length, i.e. `randomEncodedString(l - len(usernamePrefix))` instead of generating 24 bytes random string and thereafter only appending the first 24 - 13 = 11 bytes.

Reply:
It does not work as `prefix + randomEncodedString(l - len(usernamePrefix))`. base64 encoded string does not have the same length as the input string.